### PR TITLE
Cql4bonnie 508 fixes

### PR DESCRIPTION
**Pull request by lukoktonos**
This pull request is for several minor changes to update Bonnie to be 508 compliant.

Pull requests into Bonnie require the following. (Reviewer should :heavy_check_mark:*when done*):
- [X] This pull request describes *why* these changes were made.
- [X] JIRA ticket links to this PR.
    - https://jira.mitre.org/browse/BONNIE-1014
    - https://jira.mitre.org/browse/BONNIE-1015
    - https://jira.mitre.org/browse/BONNIE-1016
    - https://jira.mitre.org/browse/BONNIE-1018
    - https://jira.mitre.org/browse/BONNIE-1019
    - https://jira.mitre.org/browse/BONNIE-1020
    - https://jira.mitre.org/browse/BONNIE-1023
    - https://jira.mitre.org/browse/BONNIE-1024
- [X] JIRA test for this PR: https://jira.mitre.org/browse/BONNIE-1099
- [x] JIRA test has been added to sprint.
- [X] Automated regression test(s) pass (if possible):
    - Automated regression test passes: 
    - Before: cql4bonnie, After: cql4bonnie with these changes merged locally.

### CODE REVIEW

File: app/assets/javascripts/models/difference.js.coffee [hunk 1]
```diff
@@ -36,7 +36,18 @@ class Thorax.Collections.Differences extends Thorax.Collection
                'pass'
              else
                'fail'
-    @summary.set total: @length, matching: successful.length, percent: percent, done: done, status: status
+    # For 508 compliance, each percentage label/input pair has to have a unique ID
+    generatePercentageId = ->
```
Comment (c-monkey) [inline app/assets/javascripts/models/difference.js.coffee]: This was duplicated verbatim from `Thorax.Models.MeasureDataCriteria.generateCriteriaId` why not just use that? Or move to a shared location if needed? There is no need for duplication of code.

**Review by c-monkey — DISMISSED** [review]
https://jira.mitre.org/browse/BONNIE-1099 passes, not all things are fixed but everything called out here is.

**Review by ssayer — APPROVED** [review]
This does fix everything described in the JIRA test, however the JIRA bugs described are more far reaching than what is fixed here (mostly contrast issues). I just want to be clear about the ticket status so they aren't closed prematurely.

**Review by ssayer — CHANGES_REQUESTED** [review]
After discussing with @edeyoung, this needs to address the contrast issues on the landing page. The blue links in the footer either need to be bold or a different color

**Review by ssayer — DISMISSED** [review]
Addresses contrast issues

File: app/assets/stylesheets/builder.less [hunk 2]
```diff
@@ -208,7 +209,8 @@
           .pull-right;
         }
         &:hover {
-          background-color: @blue-darker;
+          background-color: #003456;
```
Comment (edeyoung) [inline app/assets/stylesheets/builder.less]: should this have a css variable?

File: app/assets/stylesheets/measures.less [hunk 3]
```diff
@@ -7,6 +7,7 @@
   .measure-title {
     .clearfix;
     a {
+      color: #005791;
```
Comment (edeyoung) [inline app/assets/stylesheets/measures.less]: should this have a css variable?

**Review by edeyoung — CHANGES_REQUESTED** [review]
Please respond to questions below. Otherwise, changes look good to me.

File: app/assets/stylesheets/builder.less (hunk 2 above)
Comment (edeyoung) [inline app/assets/stylesheets/builder.less]: I'm also liking `@blue-darkest` over this color: 

(blue-darkest)
image

(this color)
image

I think what is currently being used is too dark.

**Comment by edeyoung** [comment]
Looks good other than optional comments to address. jira test passes and covers issues associated with this PR.

File: app/assets/stylesheets/builder.less (hunk 2 above)
Comment (lukoktonos) [reply app/assets/stylesheets/builder.less]: Agreed--will change to blue-darkest.

File: app/assets/stylesheets/measures.less (hunk 3 above)
Comment (lukoktonos) [reply app/assets/stylesheets/measures.less]: It looks like @blue-darkest will pass the 508 spec here, so I will switch it to that.

**Review by edeyoung — APPROVED** [review]

**Pull request merged**